skyportal/plot.py

```diff
@@ -547,57 +567,684 @@ def make_custom_buttons_div(panel_name):
+        'flux': {'px': 'mjd', 'py': 'flux', 'err': 'fluxerr'},
+        'period': {'px': f'mjd_fold{ph}', 'py': 'mag', 'err': 'magerr'},
+    }
+    y_err_x = []
+    y_err_y = []
+    for d, ro in df.iterrows():
+        px = ro[err_dict[panel_name]['px']]
+        py = ro[err_dict[panel_name]['py']]
+        err = ro[err_dict[panel_name]['err']]
+
+        y_err_x.append((px, px))
+        y_err_y.append((py - err, py + err))
+    return (y_err_x, y_err_y)
+
+
+def mark_detections(plot, detection_dates, ymin, ymax):
+    """Mark detection lines on a plot
+    Parameters
+    ----------
+    plot : bokeh Figure object
+    detections_date : pandas Series object
+        The detections dates to be plotted as lines
+    ymin : int
+        Minimum y range value of the plot
+    ymax : int
+        Maximum y range value of the plot
+    Returns
+    -------
+    None
+    """
+    first = round(detection_dates.min(), 6)
+    last = round(detection_dates.max(), 6)
+    first_color = "#34b4eb"
+    last_color = "#8992f5"
+    midpoint = (ymax + ymin) / 2
+    line_top = 5 * ymax - 4 * midpoint
+    line_bottom = 5 * ymin - 4 * midpoint
+    y = np.linspace(line_bottom, line_top, num=5000)
+    first_r = plot.line(
+        x=np.full(5000, first),
+        y=y,
+        line_alpha=0.5,
+        line_color=first_color,
+        line_width=2,
+    )
+    plot.add_tools(
+        HoverTool(
+            tooltips=[("First detection", f'{first}')],
+            renderers=[first_r],
+        )
+    )
+    last_r = plot.line(
+        x=np.full(5000, last),
+        y=y,
+        line_alpha=0.5,
+        line_color=last_color,
+        line_width=2,
+    )
+    plot.add_tools(
+        HoverTool(
+            tooltips=[("Last detection", f'{last}')],
+            renderers=[last_r],
+            point_policy='follow_mouse',
+        )
+    )
+
+
+def make_legend_items_and_detection_lines(
+    grouped_data,
+    plot,
+    spinner,
+    model_dict,
+    imhover,
+    panel_name,
+    color_dict,
+    y_range,
+    data,
+    obsind,
+    markers,
+    instruments,
+    period,
+):
+    """Makes the legend items for a plot and adds detections lines.
+    grouped_data : pandas DataFrameGroupBy object
+        The photometry data grouped by label.
+    plot : bokeh Figure object
+    spinner : Spinner object
+        Spinner for controlling the data size of photometry points.
+    model_dict : dict
+        A dictionary containing all of the GlyphRenderers for the plot.
+    imhover : HoverTool object
+        HoverTool for current photometry
+    panel_name : str
+        Name of the panel. One of 'flux', 'mag' or 'period'
+    color_dict : dict
+        Dictionary defining which color the data points should be
+    y_range : tuple(int)
+        Tuple containing the minimum and maximum y range values of the plot respectively
+    data : pandas DataFrame object
+        Photometry data (ungrouped)
+    obsind : pandas Series object
+        ???
```

Comment:
If someone could help with the documentation here, that would be great. [code here](https://github.com/skyportal/skyportal/blob/master/skyportal/plot.py#L639)

Reply:
@jadalilleboe So I see
    obsind = data['hasflux'] & (
        data['flux'].fillna(0.0) / data['fluxerr'] >= PHOT_DETECTION_THRESHOLD
    )
So I would describe this as
   Index of data that exceed the required signal to noise ratio threshold
   
 What do you think @guynir42 ?

Reply:
Maybe "index of points that have been positively detected (observed), i.e., points that are not upper limits."